cmd/spire-server/cli/entry/show.go

```diff
@@ -105,127 +106,67 @@ func (s *ShowCLI) fetchEntries(ctx context.Context) error {
 		return nil
 	}
 
-	// If we didn't get any args, fetch everything
-	if s.Config.ParentID == "" && s.Config.SpiffeID == "" && len(s.Config.Selectors) == 0 {
-		err := s.fetchAllEntries(ctx)
+	filter := &entry.ListEntriesRequest_Filter{}
+	if s.Config.ParentID != "" {
+		id, err := idStringToProto(s.Config.ParentID)
 		if err != nil {
-			fmt.Printf("Error fetching entries: %s\n", err)
+			fmt.Printf("Error parsing entry parent ID %q: %v", s.Config.ParentID, err)
 			return err
 		}
-
-		return nil
+		filter.ByParentId = id
 	}
 
-	// Otherwise, fetch all records matching each constraint
-	err := s.fetchByParentID(ctx)
-	if err != nil {
-		fmt.Printf("Error fetching by parent ID: %s", err)
-		return err
-	}
-
-	err = s.fetchBySpiffeID(ctx)
-	if err != nil {
-		fmt.Printf("Error fetching by SPIFFE ID: %s", err)
-		return err
+	if s.Config.SpiffeID != "" {
+		id, err := idStringToProto(s.Config.SpiffeID)
+		if err != nil {
+			fmt.Printf("Error parsing entry SPIFFE ID %q: %v", s.Config.SpiffeID, err)
+			return err
+		}
+		filter.BySpiffeId = id
 	}
 
-	err = s.fetchBySelectors(ctx)
-	if err != nil {
-		fmt.Printf("Error fetching by selectors: %s", err)
-		return err
+	if len(s.Config.Selectors) != 0 {
+		selectors := make([]*types.Selector, len(s.Config.Selectors))
+		for i, sel := range s.Config.Selectors {
+			selector, err := parseSelector(sel)
+			if err != nil {
+				return err
+			}
+			selectors[i] = selector
+		}
+		filter.BySelectors = &types.SelectorMatch{
+			Selectors: selectors,
+			Match:     types.SelectorMatch_MATCH_SUBSET,
```

Comment:
is it expect to be always match subset?

Reply:
good catch! it should be "match exact" instead